### ISSUE TITLE
SageMath 10.3 Qepcad crashed -- automatically restarting

### ISSUE THREAD

**Issue by anirjoshi** [sagemath/sage#38302]
### Steps To Reproduce

1. I installed Sage-Math 10.3 for my Mac M2 from [https://github.com/3-manifolds/Sage_macOS/releases/](link)
2. I tried running a program containing QEPCAD
3. Sage outputs `Qepcad crashed -- automatically restarting`
4. Previously asked about this, but the response was that this is resolved in 10.3 [previous issue](https://github.com/sagemath/sage/issues/38050#issue-2308818637)
5. The input program is the following: Note that even without `memcells='10000000 +L5000'` it crashes.
```
var('x_5_0')
var('x_5_1')
var('d')
var('x_0_0')
var('x_0_1')
var('x_1_0')
var('x_1_1')
var('x_2_0')
var('x_2_1')
var('x_3_0')
var('x_3_1')
var('x_4_0')
var('x_4_1')
var('tmp')
qf = qepcad_formula
constr0 = (tmp - tmp +(Rational(-1/1)* (d)) + (Rational(1/1)* (x_0_0^2)) + (Rational(-3/4))< 0)
constr1 = (tmp - tmp +(Rational(-1/2)) + (Rational(-1/1)* (d))< 0)
constr2 = (tmp - tmp +(Rational(1/1)) + (Rational(-1/1)* (d))< 0)
constr3 = (tmp - tmp +(Rational(-1/2)) + (Rational(-1/1)* (d))< 0)
constr4 = (tmp - tmp +(Rational(1/4)) + (Rational(-1/1)* (d))< 0)
constr5 = (tmp - tmp +(Rational(-1/1)* (d)) + (Rational(-1/1)) + (Rational(1/1)* (x_5_1^2)) + (Rational(1/1)* (x_5_0^2))< 0)
constr6 = (tmp - tmp +(Rational(-1/1)* (x_0_0)) + (Rational(-1/1)* (d)) + (Rational(-1/1)* (x_0_0^2)) + (Rational(-1/4))< 0)
constr7 = (tmp - tmp +(Rational(2/1)* (x_0_0)) + (Rational(-1/1)* (d)) + (Rational(-1/1)* (x_0_0^2)) + (Rational(-9/4))< 0)
constr8 = (tmp - tmp +(Rational(1/1)* (x_0_0)) + (Rational(-1/1)* (d)) + (Rational(-1/1)* (x_0_0^2)) + (Rational(-1/4))< 0)
constr9 = (tmp - tmp +(Rational(-1/1)* (x_0_0)) + (Rational(-1/1)* (d)) + (Rational(-1/1)* (x_0_0^2)) + (Rational(-3/2))< 0)
constr10 = (tmp - tmp +(Rational(1/1)* (x_5_1)) + (Rational(-1/1)* (x_5_0^2)) + (Rational(-1/1)* (d)) + (Rational(3/4)) + (Rational(2/1)* (x_0_0)* (x_5_0)) + (Rational(-1/1)* (x_5_1^2)) + (Rational(-1/1)* (x_0_0^2))< 0)
constr11 = (tmp - tmp +(Rational(-3/2)) + (Rational(-1/1)* (d))< 0)
constr12 = (tmp - tmp +(Rational(0/1)) + (Rational(-1/1)* (d))< 0)
constr13 = (tmp - tmp +(Rational(3/4)) + (Rational(-1/1)* (d))< 0)
constr14 = (tmp - tmp +(Rational(-1/1)* (x_5_1)) + (Rational(-1/1)* (x_5_1^2)) + (Rational(1/2)) + (Rational(-1/1)* (x_5_0)) + (Rational(-1/1)* (x_5_0^2)) + (Rational(-1/1)* (d))< 0)
constr15 = (tmp - tmp +(Rational(1/2)) + (Rational(-1/1)* (d))< 0)
constr16 = (tmp - tmp +(Rational(-5/4)) + (Rational(-1/1)* (d))< 0)
constr17 = (tmp - tmp +(Rational(-2/1)* (x_5_1)) + (Rational(-1/1)* (x_5_1^2)) + (Rational(-1/1)) + (Rational(2/1)* (x_5_0)) + (Rational(-1/1)* (x_5_0^2)) + (Rational(-1/1)* (d))< 0)
constr18 = (tmp - tmp +(Rational(-1/4)) + (Rational(-1/1)* (d))< 0)
constr19 = (tmp - tmp +(Rational(-1/1)* (x_5_1)) + (Rational(-1/1)* (x_5_1^2)) + (Rational(1/2)) + (Rational(1/1)* (x_5_0)) + (Rational(-1/1)* (x_5_0^2)) + (Rational(-1/1)* (d))< 0)
constr20 = (tmp - tmp +(Rational(-2/1)* (x_5_1)) + (Rational(-1/1)* (x_5_1^2)) + (Rational(-1/4)) + (Rational(-1/1)* (x_5_0)) + (Rational(-1/1)* (x_5_0^2)) + (Rational(-1/1)* (d))< 0)
constr21 = (tmp - tmp +(Rational(-1/1)* (d))< 0)
F = qf.and_(constr0, constr1, constr2, constr3, constr4, constr5, constr6, constr7, constr8, constr9, constr10, constr11, constr12, constr13, constr14, constr15, constr16, constr17, constr18, constr19, constr20, constr21)
print(qepcad(qf.exists(x_0_0,F) ,memcells='10000000 +L5000'))
```

### Expected Behavior

It should output the formula the we get after quantifier elimination.

### Actual Behavior

Sage crashes and outputs `Qepcad crashed -- automatically restarting`

### Additional Information

Similar issues were raised previously in a lot of places including:
[link 1](https://github.com/sagemath/sage/issues/38050#issue-2308818637)
[link 2](https://github.com/sagemath/sage/issues/37349#issue-2135481233)
[link 3](https://github.com/sagemath/sage/issues/37349#issue-2135481233)
[link 4](https://github.com/sagemath/sage/issues/37518#issue-2164791030)

### Environment

```markdown
- **OS**: Mac M2 Ventura 13.4
- **Sage Version**: 10.3
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by maxale** [comment]
This can be marked as a duplicate of #37365
A close look reveals that QEPCAD in this case complains `Reason for the failure: Too few cells reclaimed.` and the issue is resolved by adding an extra `0` to the `memecells` parameter, that is, `memcells='100000000 +L5000'`

**Comment by anirjoshi** [comment]
@maxale Thank you for the pointer. But it crashes even after adding an extra `0` in the `memecells` parameter.

**Comment by maxale** [comment]
It works fine at Sagecell with `memcells='100000000 +L5000'`:
https://sagecell.sagemath.org/?q=uivsfv

**Comment by anirjoshi** [comment]
@maxale Thank you! Unfortunately, it still seems to crash in my system. However, the output `d - 1 > 0 /\ x51^2 + x50^2 - d - 1 < 0` of Sagecell doesn't seem to make sense because there is no variable with the name `x51` and `x50`.

**Comment by maxale** [comment]
Whether or not it makes sense is a different story. The point is that it works fine at Sagecell.

**Comment by anirjoshi** [comment]
The output has a variable that is not present in the input. Probably, I should file it in another issue? Anyways @maxale Thank you for the quick response.

**Comment by DaveWitteMorris** [comment]
As [explained above](https://github.com/sagemath/sage/issues/38302#issuecomment-2198244492), this can be closed as a duplicate of [#37365](https://github.com/sagemath/sage/issues/37365)
